### PR TITLE
Add cl-competiond as an entry_point to setup.py

### LINKED ISSUE #1286: add `cl-competitiond` endpoint to `setup.py`
Instead of running `scripts/competitiond.py`, install `cl-competitiond` to make it more consistent with how we invoke other endpoints.

Then we should update the docs.

### PR DESCRIPTION
**Pull request by candicegjing**
Fixed #1286

### CODE REVIEW

File: docs/Competitions.md [hunk 1]
```diff
@@ -42,13 +42,13 @@ Install all the dependencies:
 
     cd codalab-worksheets
     virtualenv -p python3.6 venv
```
Comment (percyliang) [inline docs/Competitions.md]: Don't need to git clone or make a venv anymore.  The whole point is that you can now just `pip install codalab` and type `cl-competitiond ...`

**Review by percyliang — CHANGES_REQUESTED** [review]

File: docs/Competitions.md (hunk 1 above)
Comment (percyliang) [reply docs/Competitions.md]: But should check that this works.

Comment (candicegjing) [reply docs/Competitions.md]: The only thing is that users may not have all the package requirements installed in their local machine.

Comment (percyliang) [reply docs/Competitions.md]: `pip install codalab` should install the dependencies.

Comment (candicegjing) [reply docs/Competitions.md]: That's true. I installed it in local mode when I was testing the code change. It didn't install the dependencies. I'll update the code.

**Review by percyliang — APPROVED** [review]

**Pull request merged**